tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1861,9 +1863,7 @@ def feature_importance(self, data=None, model=None, features=None, feature_stage
         if (data is None) and (not self._trainer.is_data_saved):
             raise AssertionError(
                 'No data was provided and there is no cached data to load for feature importance calculation. `cache_data=True` must be set in the `TabularPredictor` init `learner_kwargs` argument call to enable this functionality when data is not specified.')
-        if data is not None:
-            # Avoid crash when indices are duplicated
-            data = data.reset_index(drop=True)
+        self._validate_unique_indices(data, 'data')
```

Comment:
data can be None here, only validate if data is not None, otherwise will crash.

```
if data is not None:
    self._validate_unique_indices(data, 'data')
```